### DISCUSSION TITLE
-D BUILD_TIME_UNIX=$UNIX_TIME in platform.ini

### DISCUSSION THREAD

**Discussion by EDsteve** [LIFsCode/ELOC-3.0#111] [General]
@OOHehir @LIFsCode 

Hi Owen,
i know your time is limited. But i hope you can answer this question without doing much research :)
Fabi is was asking about the purpose of  -D BUILD_TIME_UNIX=$UNIX_TIME in platform.ini. It prolongs the build  time by a big amount and he was hoping to remove it IF it doesn't serve an important purpose. Do you remember what that was for?

Thanks
ED

**Comment by OOHehir** [comment]
@EDsteve @LIFsCode

Hello ED, this code generates a define (BUILD_TIME_UNIX) from the the time at code compilation. BUILD_TIME_UNIX is used to do set an initial system time & also used for some simple sanity checks, e.g. checking that the system time is not set for sometime in the past. There is a brief discussion [here](https://community.platformio.org/t/dynamic-build-flags-in-platformio-ini-unix-time/325). 

I's be surprised if this incurs any direct build overhead, but perhaps it causes files to be recompiled as it changes at every build? If required you could just substitute it with say the time [now](https://www.unixtimestamp.com/) as follows:

-D BUILD_TIME_UNIX=1734346421

This would create a constant & perhaps avoid files being recompiled every time. Every few months it would need to be updated however.

**Comment by LIFsCode** [comment]
@OOHehir @EDsteve 
Yes I got the point why you wanted to use it to initialize the time similar to what I did [here](https://github.com/LIFsCode/ELOC-3.0/blob/f059ecaed4709a43391960d2cfc7e43e225a27ef/eloc610LowPowerPartition/src/main.cpp#L388).

But setting defines in the platformio.ini affects the whole project globally, and thus everything has to be recompiled, since the compiler cannot know in advance which files use the define. The dependency management is on file basis only.

So if it is only used to init time I'll rework that.
